proposals/lip-0050.md

```diff
@@ -162,23 +279,70 @@ The `params` property of a register BLS key transaction must obey the following
 ```java
 registerBLSKeyParamsSchema = {
     "type": "object",
-    "required": ["blsKey", "proofOfPossession"],
+    "required": ["blsKey", "proofOfPossession", "generatorKey"],
     "properties": {
         "blsKey": {
             "dataType": "bytes",
+            "length": 48,
             "fieldNumber": 1
         },
         "proofOfPossession": {
             "dataType": "bytes",
+            "length": 96,
             "fieldNumber": 2
+        },
+        "generatorKey": {
+            "dataType": "bytes",
+            "length": 32,
+            "fieldNumber": 3
         }
     }
 }
 ```
 
+#### Verification
+
+```python
+def verify(trs: Transaction) -> None:
+
+    validatorAddress = 20-byte address derived from trs.senderPublicKey
+
+    if validator.getValidatorAccount(validatorAddress) does not exists :
+        raise Exception('Public key does not correspond to a registered validator.')
+
+    if validatorAccount.blsKey != INVALID_BLS_KEY:
+        raise Exception('Validator already has a registered keys.')
+```
+
 ##### Execution
 
-Executing a transaction `trs` triggering an register BLS key command is done by calling the function `setValidatorBLSKey(validatorAddress, trs.params.proofOfPossession, trs.params.blsKey)` where `validatorAddress` is the 20-byte address derived from `trs.senderPublicKey` and `setValidatorBLSKey` is defined in the [Validators module][lip-0044#setValidatoBLSKey]. If this function returns false, the transaction is invalid.
+```python
+def execute(trs: Transaction) -> None:
+    validatorAddress = 20-byte address derived from trs.senderPublicKey
+
+    validator.setValidatorGeneratorKey(validatorAddress, trs.params.generatorKey)
+
+    # the calls below will raise an exception if the proof of possession is invalid
+    # with respect to the given BLS key.
+    validator.setValidatorBLSKey(validatorAddress, trs.params.proofOfPossession, trs.params.blsKey)
+    emitEvent(
```

Comment:
An event would be emitted by the validator module when the setting of the BLS key fails, this event should be a persistent event. So that it will stay in the event tree.
In that case, the command has no effect and no keys will be set.